OpenRA.Mods.Common/Scripting/Global/WeatherGlobal.cs

```diff
@@ -0,0 +1,165 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Scripting;
+
+namespace OpenRA.Mods.Common.Scripting.Global
+{
+	[ScriptGlobal("Weather")]
+	public class WeatherGlobal : ScriptGlobal
+	{
+		readonly WeatherOverlay overlay;
+		readonly bool hasAnOverlay;
+
+		public WeatherGlobal(ScriptContext context)
+			: base(context)
+		{
+			overlay = context.World.WorldActor.TraitOrDefault<WeatherOverlay>();
+			hasAnOverlay = overlay != null;
+		}
+
+		public bool Active
+		{
+			get { return hasAnOverlay && overlay.Active; }
+			set { if (hasAnOverlay) overlay.Active = (bool)value; }
+		}
+
+		[Desc("Average number of particles per 1000x1000 px square.")]
+		public int ParticleDensityFactor
+		{
+			get { return hasAnOverlay ? (int)(overlay.ParticleDensityFactor) : 0; }
+			set
+			{
+				if (!hasAnOverlay)
+					return;
+
+				overlay.ParticleDensityFactor = (int)value;
+				overlay.CalculateMaxParticleCount();
+			}
+		}
+
+		[Desc("Should the level of the wind change over time, or just stick to the first value of WindLevels?")]
+		public bool ChangingWindLevel
```

Comment:
Directly mapping the trait's properties into lua will make it much more difficult if we want to change the way the trait works in the future (which we know that we need to, e.g. wind).